Failure to package on dotnet sdk > 3.1.102

Packaging seems to work when using dotnet msbuild /t:CreateDeb..... but NOT when running dotnet deb .....

dotnet deb runs fine on any dotnet 3.1 sdk <= 3.1.102, but any newer versions are throwing the following errors:

```
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: The "ProcessFrameworkReferences" task failed unexpectedly.
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'NuGet.Frameworks, Version=5.5.0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: File name: 'NuGet.Frameworks, Version=5.5.0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at System.Linq.Enumerable.WhereEnumerableIterator`1.ToList()
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ProcessFrameworkReferences.ExecuteCore()
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute()
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:
```

System Specs:
Ubuntu 18.04
Kernel 5.3.0-45-generic
dotnet version: 3.1.201
dotnet-deb version: 0.1.155+9e4041d64a

Have similar problem on Win7 x64 (Russian)+Visual Studio 2019.
Probably localized error messages can be translated by Google (but it's not need, the errors are same as areis422 said).

> D:\WebOne>dotnet zip -c Release
> dotnet zip (0.1.155+9e4041d64a)
> __________________________________________________
> Проект "D:\WebOne\WebOne.csproj" (целевые объекты Restore):
> 
> __________________________________________________
> Проект "D:\WebOne\WebOne.csproj" создает "D:\WebOne\WebOne.csproj"
>  (целевые объекты _GenerateRestoreGraphProjectEntry):
> 
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: непредвиденная о
> шибка при выполнении задачи ProcessFrameworkReferences".
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: System.IO.FileNo
> tFoundException: Could not load file or assembly 'NuGet.Frameworks, Version=5.5.
> 0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Не удается найти указанн
> ый файл.
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018: File name: 'NuGe
> t.Frameworks, Version=5.5.0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
> 
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at System.Lin
> q.Enumerable.WhereEnumerableIterator`1.ToList()
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at System.Lin
> q.Enumerable.ToList[TSource](IEnumerable`1 source)
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.
> NET.Build.Tasks.ProcessFrameworkReferences.ExecuteCore()
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.
> NET.Build.Tasks.TaskBase.Execute()
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.
> Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execu
> te()
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:    at Microsoft.
> Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecuti
> onHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket buc
> ket, TaskExecutionMode howToExecuteTask)
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:
> C:\Program Files\dotnet\sdk\3.1.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET
> .Sdk.FrameworkReferenceResolution.targets(59,5): error MSB4018:
> Сборка проекта "WebOne.csproj" завершена с ошибкой.
> Failed to restore 'WebOne.csproj'. Please run dotnet restore, and try again.
> 
> D:\WebOne>dotnet restore
>   Восстановление завершено в 60,22 ms для D:\WebOne\WebOne.csproj.
> 
> D:\WebOne>dotnet zip --no-restore -c Release
> dotnet zip (0.1.155+9e4041d64a)
> Microsoft (R) Build Engine версии 16.5.0+d4cbfca49 для .NET Core
> (C) Корпорация Майкрософт (Microsoft Corporation). Все права защищены.
> 
>   WebOne -> D:\WebOne\bin\Release\netcoreapp3.1\WebOne.dll
>   WebOne -> D:\WebOne\bin\Release\netcoreapp3.1\publish\
>   Creating zip package D:\WebOne\bin\Release\netcoreapp3.1\WebOne.0.10.0.zip
> 
> D:\WebOne>

--- a/.azure-pipelines.yml
+++ b/.azure-pipelines.yml
@@ -9,6 +9,10 @@ resources:
     image: mcr.microsoft.com/dotnet/core/sdk:3.0-disco
   - container: 3.0-buster
     image: mcr.microsoft.com/dotnet/core/sdk:3.0-buster
+  - container: 3.1-focal
+    image: mcr.microsoft.com/dotnet/core/sdk:3.1-focal
+  - container: 5.0-focal
+    image: mcr.microsoft.com/dotnet/core/sdk:5.0-focal
 
 stages:
 - stage: Build
@@ -60,6 +64,32 @@ stages:
         3.0-disco-tarball:
           container: 3.0-disco
           command: tarball
+
+        3.1-focal-deb:
+          container: 3.1-focal
+          command: deb
+        3.1-disco-rpm:
+          container: 3.1-focal
+          command: rpm
+        3.1-focal-zip:
+          container: 3.1-focal
+          command: zip
+        3.1-focal-tarball:
+          container: 3.1-focal
+          command: tarball
+
+        # 5.0-focal-deb:
+        #   container: 5.0-focal
+        #   command: deb
+        # 5.0-disco-rpm:
+        #   container: 5.0-focal
+        #   command: rpm
+        # 5.0-focal-zip:
+        #   container: 5.0-focal
+        #   command: zip
+        # 5.0-focal-tarball:
+        #   container: 5.0-focal
+        #   command: tarball
 
     container: $[ variables['container'] ]
     steps:

--- a/.azure-pipelines.yml
+++ b/.azure-pipelines.yml
@@ -114,8 +114,11 @@ stages:
         pip3 install --upgrade --user setuptools
         pip3 install --user molecule
         pip3 install --user docker-py
+        pip3 install --user pytest
+        pip3 install --user testinfra
         molecule --version
         ansible --version
+        pytest --version
       displayName: Install molecule
     - task: DownloadBuildArtifacts@0
       displayName: Download Build Artifacts

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,7 @@ env:
   - RUNTIME=ubuntu.14.04-x64
   - RUNTIME=ubuntu.16.04-x64
   - RUNTIME=ubuntu.18.04-x64
-  - RUNTIME=ubuntu.19.04-x64
+  - RUNTIME=ubuntu.19.10-x64
   - RUNTIME=opensuse.42.3-x64
 
 sudo: required

--- a/dotnet-rpm/PackagingRunner.cs
+++ b/dotnet-rpm/PackagingRunner.cs
@@ -6,6 +6,7 @@ using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Runtime.Loader;
 using System.Text;
 using ConsoleLogger = Microsoft.Build.Logging.ConsoleLogger;
 using IMSBuildLogger = Microsoft.Build.Framework.ILogger;
@@ -22,7 +23,19 @@ namespace Dotnet.Packaging
 
         public PackagingRunner(string outputName, string msbuildTarget, string commandName)
         {
-            MSBuildLocator.RegisterDefaults();
+            var instance =  MSBuildLocator.RegisterDefaults();
+
+            // Workaround for https://github.com/microsoft/MSBuildLocator/issues/86
+            AssemblyLoadContext.Default.Resolving += (assemblyLoadContext, assemblyName) =>
+            {
+                var path = Path.Combine(instance.MSBuildPath, assemblyName.Name + ".dll");
+                if (File.Exists(path))
+                {
+                    return assemblyLoadContext.LoadFromAssemblyPath(path);
+                }
+
+                return null;
+            };
 
             this.outputName = outputName;
             this.msbuildTarget = msbuildTarget;

--- a/molecule/framework-dependent/molecule/default/molecule.yml
+++ b/molecule/framework-dependent/molecule/default/molecule.yml
@@ -3,8 +3,9 @@ dependency:
   name: galaxy
 driver:
   name: docker
-lint:
-  name: yamllint
+lint: |
+  set -e
+  yamllint .
 platforms:
   # This list should be roughly kept in sync with the list of
   # OSes for which .NET Core builds packages, as seen at
@@ -35,8 +36,9 @@ platforms:
     image: opensuse/leap:15.0
 provisioner:
   name: ansible
-  lint:
-    name: ansible-lint
+  lint: |
+    set -e
+    ansible-lint
   inventory:
     host_vars:
       fedora.30:
@@ -46,5 +48,6 @@ scenario:
   name: default
 verifier:
   name: testinfra
-  lint:
-    name: flake8
+  lint: |
+    set -e
+    flake8

--- a/molecule/framework-dependent/molecule/default/molecule.yml
+++ b/molecule/framework-dependent/molecule/default/molecule.yml
@@ -20,8 +20,8 @@ platforms:
     image: ubuntu:16.04
   - name: ubuntu.18.04
     image: ubuntu:18.04
-  - name: ubuntu.19.04
-    image: ubuntu:19.04
+  - name: ubuntu.19.10
+    image: ubuntu:19.10
   - name: debian.9
     image: debian:9
   - name: debian.10

--- a/molecule/self-contained/molecule/default/molecule.yml
+++ b/molecule/self-contained/molecule/default/molecule.yml
@@ -3,8 +3,9 @@ dependency:
   name: galaxy
 driver:
   name: docker
-lint:
-  name: yamllint
+lint: |
+  set -e
+  yamllint .
 platforms:
   # This list should be roughly kept in sync with the list of supported OSes described at
   # https://github.com/dotnet/core/blob/master/release-notes/3.0/3.0-supported-os.md,
@@ -57,8 +58,9 @@ platforms:
 
 provisioner:
   name: ansible
-  lint:
-    name: ansible-lint
+  lint: |
+    set -e
+    ansible-lint
   inventory:
     host_vars:
       fedora.30:
@@ -70,5 +72,6 @@ verifier:
   name: testinfra
   options:
     junitxml: pytest.xml
-  lint:
-    name: flake8
+  lint: |
+    set -e
+    flake8

